source/base/mpi.cc

```diff
@@ -501,10 +516,7 @@ namespace Utilities
       // when running PETSc, because we initialize MPI ourselves before calling
       // PetscInitialize
 #ifdef DEAL_II_WITH_MPI
-      int MPI_has_been_started = 0;
-      MPI_Initialized(&MPI_has_been_started);
-      if (Utilities::System::job_supports_mpi() == true &&
-          MPI_has_been_started != 0)
+      if (job_supports_mpi() == true)
```

Comment:
We use this style of writing it as a comparison (instead of an expression) everywhere, pretty much uniformly. Of course, we can argue whether that's good or bad or right or wrong (and of course we're all very much aware of the fact that in C++, comparisons are also expressions, and all that's required here is an expression that's convertible to bool), but at least we're pretty consistent.